shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/presentation/components/RegisterScreen.kt

```diff
@@ -0,0 +1,345 @@
+                }
+
+                Text(UiText.StringRes(MR.strings.register_birthday_title_text).asString())
+                Text(text = state.birthDate,
+                    color = MaterialTheme.colorScheme.onSecondaryContainer,
+                    modifier = Modifier.clip(RoundedCornerShape(25.dp))
+                        .background(MaterialTheme.colorScheme.secondaryContainer).height(56.dp)
+                        .fillMaxWidth().padding(16.dp).clickable {
+                            onEvent(RegisterEvent.ShowBirthdayDialog)
+                        })
+
+                Divider()
+                // ADDRESS DETAILS
+                Text(
+                    text = UiText.StringRes(MR.strings.register_address_title_text).asString(),
+                    style = Typography.titleMedium,
+                    color = Color.Black.copy(alpha = 0.6f)
+                )
+
+                Spacer(modifier = Modifier.height(8.dp))
+
+                UserInput(
+                    state = state.streetName,
+                    titleText = UiText.StringRes(MR.strings.register_street_name_title_text)
+                        .asString(),
+                    placeholderText = UiText.StringRes(MR.strings.register_street_name_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(RegisterEvent.StreetNameChanged(it))
+                }
+
+                UserInput(
+                    state = if (state.doorNumber == DEFAULT_VALUE_UNFILLED) "" else state.doorNumber.toString(),
+                    titleText = UiText.StringRes(MR.strings.register_door_number_title_text)
+                        .asString(),
+                    placeholderText = UiText.StringRes(MR.strings.register_door_number_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Number, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(RegisterEvent.DoorNumberChanged(it.toInt()))
+                }
+
+                UserInput(
+                    state = state.city,
+                    titleText = UiText.StringRes(MR.strings.register_city_title_text).asString(),
+                    placeholderText = UiText.StringRes(MR.strings.register_city_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(RegisterEvent.CityChanged(it))
+                }
+
+                UserInput(
+                    state = if (state.postalCode == DEFAULT_VALUE_UNFILLED) "" else state.postalCode.toString(),
```

Comment:
Hint: if you set postalCode as nullable you won't need this condition :)  
If it is a nullable variable you can use it like this:
state.postalCode?.toString().orEmpty() 

You don't have to modify it it is just a tip :)

Reply:
Made it nullable